staging/src/k8s.io/api/OWNERS

```diff
@@ -48,3 +48,5 @@ reviewers:
 - yifan-gu
 - yujuhong
 - zmerlynn
+labels:
+- sig/api-machinery
```

Comment:
While https://github.com/kubernetes/community/blob/master/sig-api-machinery/README.md#subprojects lists `k8s.io/api`, my understanding is that individual groups/types are owned by the respective SIG.

That's the reason I didn't add this in: https://github.com/kubernetes/community/blob/master/sig-api-machinery/README.md#subprojects

/cc @liggitt 
because you are a api-approver

Reply:
There's a small amount of code here, so it probably demands a SIG, and this seems right to me